test/new-e2e/tests/windows/common/activedirectory/provisioner.go

```diff
@@ -0,0 +1,74 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package activedirectory
+
+import (
+	"fmt"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/optional"
+	"github.com/DataDog/test-infra-definitions/components/os"
+	"github.com/DataDog/test-infra-definitions/resources/aws"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/ec2"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/fakeintake"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+const (
+	provisionerBaseID = "aws-ec2vm-active-directory-"
+	defaultVMName     = "dcvm"
+)
+
+// Provisioner creates an Active Directory environment on a given VM.
+func Provisioner(opts ...ProvisionerOption) e2e.TypedProvisioner[Env] {
```

Comment:
could you use the same model than here
https://github.com/DataDog/datadog-agent/blob/main/test/new-e2e/pkg/environments/aws/host/host.go#L151-L243

func GetProvisionerParams()
func Run()
func Provisioner()

This useful to re-use or extend other deployments